src/components/Routing.tsx

```diff
@@ -38,35 +48,67 @@ export const InitialisedRouter = ({ onNavChanged }: Pick<AppRootProps, 'onNavCha
-        <SceneRedirecter />
-      </Route>
-      <Route exact path={getRoute(ROUTES.Home)}>
-        <SceneHomepage />
-      </Route>
-      <Route path={getRoute(ROUTES.Scene)}>
-        <SceneRedirecter />
-      </Route>
-      <Route path={getRoute(ROUTES.Checks)}>
-        <CheckRouter />
-      </Route>
-      <Route path={getRoute(ROUTES.Probes)}>
-        <ProbeRouter />
-      </Route>
-      <Route exact path={getRoute(ROUTES.Alerts)}>
-        <AlertingPage />
-      </Route>
-      <Route path={getRoute(ROUTES.Config)}>
-        <ConfigPage initialized />
+    <Routes>
+      <Route index element={<Navigate to={ROUTES.Home} />} />
+
+      <Route path={ROUTES.Home} element={<SceneHomepage />} />
+
+      <Route path={ROUTES.Checks}>
+        <Route index element={<CheckList />} />
+        <Route path=":id">
+          <Route index element={<DashboardPage />} />
+          <Route path="edit" element={canEdit ? <EditCheck /> : <Navigate to=".." replace />} />
+          <Route path="dashboard" element={<Navigate to=".." replace />} />
+          <Route path="*" element={<CheckNotFound />} />
+        </Route>
+        <Route path="choose-type" element={<ChooseCheckGroup />} />
+        <Route path="new">
+          <Route index element={<ChooseCheckGroup />} />
+          {getNewCheckTypeRedirects().map(({ checkType, checkTypeGroupUrl }) => (
+            <Route key={checkType} path={checkType} element={<Navigate to={`../${checkTypeGroupUrl}`} replace />} />
+          ))}
+          <Route path=":checkTypeGroup" element={<NewCheck />} />
+          <Route path="*" element={<CheckNotFound />} />
+        </Route>
+
+        <Route path="edit/:id" element={<LegacyEditRedirect entity="check" />} />
+
+        <Route
+          path="*"
+          element={
+            <PluginPageNotFound>
+              The page you are looking for does not exist. Here is a working link to{' '}
+              <TextLink href={getRoute(ROUTES.Checks)}>checks listing</TextLink>.
+            </PluginPageNotFound>
+          }
+        />
       </Route>
 
-      <Route>
-        <Redirect to={getRoute(ROUTES.Home)} />
+      <Route path={ROUTES.Probes}>
+        <Route index element={<Probes />} />
+        <Route path="new" element={<NewProbe />} />
+        <Route path=":id">
+          <Route index element={<EditProbe readOnly />} />
+          <Route path="edit" element={<EditProbe />} />
+        </Route>
+
+        <Route path="edit/:id" element={<LegacyEditRedirect entity="probe" />} />
       </Route>
-    </Switch>
+
+      <Route path={ROUTES.Alerts} element={<AlertingPage />} />
+
+      <Route path={ROUTES.Config} element={<ConfigPage initialized />} />
+
+      <Route path={ROUTES.Redirect} element={<SceneRedirecter />} />
+
+      <Route path={ROUTES.Scene} element={<SceneRedirecter />} />
+
+      {/* TODO: Create 404 instead of navigating to home(?) */}
+      <Route path="*" element={<Navigate to={`../${ROUTES.Home}`} replace />} />
```

Comment:
Wouldn't this work as a 404 page?

```suggestion
       <Route
          path="*"
          element={
            <PluginPageNotFound>
              The page you are looking for does not exist. Here is a working link to{' '}
              <TextLink href={getRoute(ROUTES.Checks)}>checks listing</TextLink>.
            </PluginPageNotFound>
          }
        />
```

Reply:
Sure thing 👍